src/Service/Telemetry/ApplicationInsightsLoggerProvider.cs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ApplicationInsights;
+using Microsoft.Extensions.Logging;
+
+/// <summary>
+/// Provides a custom logger implementation that sends telemetry events to Azure Application Insights.
+/// See <see href="https://learn.microsoft.com/en-us/dotnet/core/extensions/custom-logging-provider">Custom Logging Provider</see> for more information.
+/// </summary>
+public class ApplicationInsightsLoggerProvider : ILoggerProvider
```

Comment:
Why are you creating your own?  ApplicationInsightsLoggerProvider and ApplicationInsightsLogger are provided by the nuget package. See source: https://github.com/microsoft/ApplicationInsights-dotnet/tree/main/NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/Logging/Implementation

Reply:
They're obsolete, also we wanted to customize it for our use case, therefore not using those.

Reply:
I did more digging and they do not appear to be obsolete. In fact, `AddApplicationInsights()` uses the library provided `ApplicationInsightsLoggerProvider`. Source code: https://github.com/microsoft/ApplicationInsights-dotnet/blob/2.21.0/LOGGING/src/ILogger/ApplicationInsightsLoggerProvider.cs

Reply:
removed created `ApplicationInsightsLoggerProvider` Class.